tpl/template_funcs.go

```diff
@@ -1676,32 +1695,36 @@ func init() {
 		"partial":      partial,
 		"pluralize":    pluralize,
 		"readDir":      readDir,
-		"ref":          ref,
-		"relURL":       func(a string) template.HTML { return template.HTML(helpers.RelURL(a)) },
-		"relref":       relRef,
-		"replace":      replace,
-		"replaceRE":    replaceRE,
-		"safeCSS":      safeCSS,
-		"safeHTML":     safeHTML,
-		"safeJS":       safeJS,
-		"safeURL":      safeURL,
-		"sanitizeURL":  helpers.SanitizeURL,
-		"sanitizeurl":  helpers.SanitizeURL,
-		"seq":          helpers.Seq,
-		"sha1":         sha1,
-		"shuffle":      shuffle,
-		"singularize":  singularize,
-		"slice":        slice,
-		"slicestr":     slicestr,
-		"sort":         sortSeq,
-		"split":        split,
-		"string":       func(v interface{}) string { return cast.ToString(v) },
-		"sub":          func(a, b interface{}) (interface{}, error) { return helpers.DoArithmetic(a, b, '-') },
-		"substr":       substr,
-		"title":        func(a string) string { return strings.Title(a) },
-		"trim":         trim,
-		"upper":        func(a string) string { return strings.ToUpper(a) },
-		"urlize":       helpers.URLize,
-		"where":        where,
+		"readFile": func(i interface{}) (string, error) {
+			// TODO(bep): Stupid type assertion, see https://github.com/spf13/afero/issues/76
+			return readFile(afero.NewBasePathFs(hugofs.SourceFs, viper.GetString("WorkingDir")).(*afero.BasePathFs), cast.ToString(i))
+		},
```

Comment:
As I said, this is work in progress. This construction is to make it testable -- but I wasn't entirely happy about it, so I'm still thinking.